### PR TITLE
[improve][broker] Make some methods in TenantsBase async.

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation
See PIP #14365  and change tracker #15043. 

 Make `getTenants` /  `getTenantAdmin` / `createTenant` / `updateTenant` / `deleteTenant` methods in TenantsBase to pure async.

### Documentation

- [x] `no-need-doc` 
(Please explain why)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java [hunk 1]
```diff
@@ -90,23 +84,20 @@ public void getTenants(@Suspended final AsyncResponse asyncResponse) {
...
-            }
-            boolean response = tenantInfo.isPresent() ? asyncResponse.resume(tenantInfo.get())
-                    : asyncResponse.resume(new RestException(Status.NOT_FOUND, "Tenant does not exist"));
-            return;
-        });
+        validateSuperUserAccessAsync()
+                .thenCompose(__ -> tenantResources().getTenantAsync(tenant))
+                .thenApply(tenantInfo -> {
+                    if (!tenantInfo.isPresent()) {
+                        new RestException(Status.NOT_FOUND, "Tenant does not exist");
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:91]: miss `throw`.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java [hunk 2]
```diff
@@ -90,23 +84,20 @@ public void getTenants(@Suspended final AsyncResponse asyncResponse) {
...
+                    }
+                    return tenantInfo.get();
+                })
+                .thenAccept(asyncResponse::resume)
+                .exceptionally(ex -> {
+                    log.error("[{}] Failed to get tenant admin {}", clientAppId(), ex);
```
Comment (nodece) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:97]: If the tenant does not exist, we don't need to log an error.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java (hunk 1 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:91]: Oh, yes. fixed.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java (hunk 2 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:97]: Keep the original behavour.

**Review by nodece — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java [hunk 3]
```diff
@@ -63,24 +63,18 @@ public class TenantsBase extends PulsarWebResource {
...
-            deepCopy.sort(null);
-            asyncResponse.resume(deepCopy);
-        });
+        validateSuperUserAccessAsync()
+                .thenCompose(__ -> tenantResources().listTenantsAsync())
+                .thenAccept(tenants -> {
+                    // deep copy the tenants to avoid concurrent sort exception
+                    List<String> deepCopy = new ArrayList<>(tenants);
+                    deepCopy.sort(null);
+                    asyncResponse.resume(deepCopy);
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:72]:
```suggestion
                    asyncResponse.resume(Collections.unmodifiableListdeepCopy(deepCopy));
```

**Review by codelipenghui — APPROVED** [review]

**Review by congbobo184 — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java [hunk 4]
```diff
@@ -63,24 +63,18 @@ public class TenantsBase extends PulsarWebResource {
...
+                }).exceptionally(ex -> {
+                    log.error("[{}] Failed to get tenants list", clientAppId(), ex);
```
Comment (gaoran10) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:74]: I'm not sure if the `HttpRequest` object will be reused in async way, if it reused the client appid may be modified.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java (hunk 3 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:72]: Ah, this seems ok, we return a new list to the user, so we don't care the result, right ?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java (hunk 4 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TenantsBase.java:74]: Got it. I will define this at the beginning.

**Review by HQebupt — APPROVED** [review]

**Review by shibd — APPROVED** [review]

**Review by mattisonchao — APPROVED** [review]

**Pull request merged**